lib/backend/k8s/resources/workloadendpoint.go

```diff
@@ -60,64 +60,78 @@ func (c *WorkloadEndpointClient) Create(ctx context.Context, kvp *model.KVPair)
 	// Note: it's a bit odd to do this in the Create, but the CNI plugin uses CreateOrUpdate().  Doing it
 	// here makes sure that, if the update fails: we retry here, and, we don't report success without
 	// making the patch.
-	return c.patchPodIP(ctx, kvp)
+	return c.patchInPodIPs(ctx, kvp)
 }
 
 func (c *WorkloadEndpointClient) Update(ctx context.Context, kvp *model.KVPair) (*model.KVPair, error) {
 	log.Debug("Received Update request on WorkloadEndpoint type")
 	// As a special case for the CNI plugin, try to patch the Pod with the IP that we've calculated.
 	// This works around a bug in kubelet that causes it to delay writing the Pod IP for a long time:
 	// https://github.com/kubernetes/kubernetes/issues/39113.
-	return c.patchPodIP(ctx, kvp)
+	return c.patchInPodIPs(ctx, kvp)
 }
 
 func (c *WorkloadEndpointClient) DeleteKVP(ctx context.Context, kvp *model.KVPair) (*model.KVPair, error) {
 	return c.Delete(ctx, kvp.Key, kvp.Revision, kvp.UID)
 }
 
 func (c *WorkloadEndpointClient) Delete(ctx context.Context, key model.Key, revision string, uid *types.UID) (*model.KVPair, error) {
-	log.Warn("Operation Delete is not supported on WorkloadEndpoint type")
-	return nil, cerrors.ErrorOperationNotSupported{
-		Identifier: key,
-		Operation:  "Delete",
-	}
+	log.Debug("Delete for WorkloadEndpoint, patching out annotations.")
+	return c.patchOutPodIPs(ctx, key, revision, uid)
 }
 
-// patchPodIP PATCHes the Kubernetes Pod associated with the given KVPair with the IP addresses it contains.
+// patchInPodIPs PATCHes the Kubernetes Pod associated with the given KVPair with the IP addresses it contains.
 // This is a no-op if there is no IP address.
 //
 // We store the IP addresses in annotations because patching the PodStatus directly races with changes that
 // kubelet makes so kubelet can undo our changes.
-func (c *WorkloadEndpointClient) patchPodIP(ctx context.Context, kvp *model.KVPair) (*model.KVPair, error) {
+func (c *WorkloadEndpointClient) patchInPodIPs(ctx context.Context, kvp *model.KVPair) (*model.KVPair, error) {
 	ips := kvp.Value.(*apiv3.WorkloadEndpoint).Spec.IPNetworks
 	if len(ips) == 0 {
 		return kvp, nil
 	}
 
 	log.Debugf("PATCHing pod with IPs: %v", ips)
-	wepID, err := c.converter.ParseWorkloadEndpointName(kvp.Key.(model.ResourceKey).Name)
+	key := kvp.Key
+	return c.patchPodIPAnnotations(key, kvp.Revision, kvp.UID, ips)
+}
+
+// patchOutPodIPs sets our pod IP annotations to empty strings; this is used to signal that the IP has been removed
+// from the pod at teardown.
+func (c *WorkloadEndpointClient) patchOutPodIPs(ctx context.Context, key model.Key, revision string, uid *types.UID) (*model.KVPair, error) {
+	return c.patchPodIPAnnotations(key, revision, uid, nil)
```

Comment:
Might be worth a follow on PR to remove this section of the CNI plugin, since I think it's dead code with this change? https://github.com/projectcalico/cni-plugin/blob/0c2676672f697b1ce3ac1ad5f27b57135fb5395c/pkg/k8s/k8s.go#L509

Not necessarily a problem, but could be confusing in the future when we don't remember why it's there :D